integration/test/cptcconnectivity/cptcconnectivity_test.go

```diff
@@ -0,0 +1,109 @@
+// +build k8srequired
+
+package cptcconnectivity
+
+import (
+	"context"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/azure-operator/v3/integration/env"
+)
+
+func Test_Connectivity(t *testing.T) {
+
+	err := connectivity.Test(context.Background())
+	if err != nil {
+		t.Fatalf("%#v", err)
+	}
+}
+
+type Config struct {
+	ClusterID string
+	Logger    micrologger.Logger
+	G8sClient versioned.Interface
+	K8sClient kubernetes.Interface
+}
+
+type Connectivity struct {
+	clusterID string
+	logger    micrologger.Logger
+	g8sClient versioned.Interface
+	k8sClient kubernetes.Interface
+}
+
+func New(config Config) (*Connectivity, error) {
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+
+	s := &Connectivity{
+		clusterID: config.ClusterID,
+		logger:    config.Logger,
+		g8sClient: config.G8sClient,
+		k8sClient: config.K8sClient,
+	}
+
+	return s, nil
+}
+
+func (s *Connectivity) Test(ctx context.Context) error {
+	s.logger.LogCtx(ctx, "level", "debug", "message", "testing connectivity between control plane cluster and tenant cluster")
+	podName := "e2e-connectivity"
+	podNamespace := "default"
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: podNamespace,
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers:    []v1.Container{{Name: "connectivity", Image: "busybox", Command: []string{"nc"}, Args: []string{"-z", "api." + s.clusterID + ".k8s." + env.CommonDomain(), "443"}}},
```

Comment:
To test the connectivity, the pod uses `netcat` and tries to connect to the tenant cluster API.